### ISSUE TITLE
SLEAP training stopped predicting frames

### ISSUE THREAD

**Issue by rubytseng18** [talmolab/sleap#2374]
Hi, 

I have been running into issues while training my model. It would train for both the centroid and centered instance, but then re-run the centered instance. Then it would say that it predicted on 0 frames. I haven't had this issue before, and the only difference I made was labeling more frames. 

**These are the details regarding the behaviour set up and model:** 

**Behaviour set up**: 1 single black mouse with a Neuralpixel scope. The wire is also in frame. 

<img width="1408" height="884" alt="Image" src="https://github.com/user-attachments/assets/7bd63f5d-9240-4f0b-b30f-302106e02185" />

**Model:** a multi-animal top-down 

<img width="862" height="594" alt="Image" src="https://github.com/user-attachments/assets/24d13773-d482-4c58-a19e-4e1506ba206e" />

<img width="875" height="595" alt="Image" src="https://github.com/user-attachments/assets/5dd3e5a3-a4d3-44c6-9c98-12b5bcea9787" />

The most recent successfully run model statistics: 

<img width="1081" height="432" alt="Image" src="https://github.com/user-attachments/assets/68e6bef9-080b-4982-b953-fde6547b3322" />

I decided to add more frames and ever since then it wouldn't let run and would encounter the problem where it would run and predict on 0 frames. I tried only predicting without training and it would still not work. 

The error code from the miniforge prompt is attached [here](https://drive.google.com/file/d/1Iz1VDjqrWilvePau5XKO24kPix0uH7Tb/view?usp=sharing): 

The error code from miniforge prompt after training using single mouse [here](https://drive.google.com/file/d/13hnbRj84Jd0Smvw4hhlDecMQzfyR3FBE/view?usp=sharing): 

I would appreciate any tips and advice on how to continue to train and improve the model. Please let me know if you need more information from me. 

Thank you so much!

**Comment by eberrigan** [comment]
Hello @rubytseng18,

It sounds like the centroid model training might have failed to find your animal. Can you also show screenshots of the receptive field size? We have some example notebooks to help with that -- https://github.com/talmolab/sleap-nn/tree/main/example_notebooks.

Thanks!

Elizabeth

**Comment by rpainaik** [comment]
Hi! I'm having issues with this as well. Were you able to find a solution?

**Comment by amadorkane** [comment]
Same here!  Any solutions or advice? I am tracking a single spider in dorsal view.  With up to 85 frames labeled (all tarsi, head, caudal and and COM), being careful to not label any occluded or unclear keypoints, I had been able to get very accurate predictions for all points using multianimal with a single instance.  After labeling an additional 20 frames, which visually appear similar to the originals (no changes in body morphology, intensity, background, etc.), SLEAP either fails to predict altogether (0 predictions) or produces predictions for all points centered on top of the COM.  I'm using the same training hyperparameters as before.  Pretty baffled!

**Comment by amadorkane** [comment]
FYI, I was able get it to train and predict well again after reverting to an earlier project in which I had only labeled 75 frames in a single video. Goodbye those additional 30 labeled frames!  I'm now adding frames a few at a time and retraining/predicting each round, being careful to save and record which projects and models work. Still don't know what about my labeled frames made it mess up, but this strategy worked for me so you might try.

**Comment by amadorkane** [comment]
Another update:  the exact same thing happened with a second video:  I was able to get accurate predictions after labeling about 80 frames, then adding more suddenly made it stop predicting or predict wildly inaccurate labels. Once again I'm being super careful about labeling.  I have tried the usual tricks to avoid overfitting (trying different batch sizes and epoch numbers, using large enough receptive field, etc.)  It's able to find the centroid just fine.  The issue is having it repeatedly fall apart for the centered instance labels when I try to enlarge the training set.

**Comment by alicup29** [comment]
Hi all and apologies for the delay,

If I understand correctly, the pattern is that: a top-down model (centroid + centered-instance) that was training and predicting fine suddenly starts reporting 0 predicted frames, or predicts junk centered on the centroid after adding more labeled frames to the project.

There's a couple of things worth trying and sharing to help us tackle the issue:

1. After adding your new frames, try re-running the `sleap-nn` config picker on the updated project and check whether the recommended `max_stride` and/or `crop_size` changed compared to your last working config. If it grew compared to your original training config, try retraining with the updated value. For example, `sleap-nn config labels.slp`, or `sleap-nn config --help` for more options.
2. @amadorkane's workaround is a good fallback: add labeled frames a few at a time, retraining/predicting after each small batch, so you can pin down exactly which frame(s) trigger the collapse. If a specific frame is the trigger, it's worth double-checking that frame's labels for a stray or occluded point placed far from the animal, since a single outlier point can blow up the effective bounding box used to size the crop.
3. To help us dig in further on your specific case if the above isn't working out, could you share:
   - The output of `sleap doctor` (copy-paste the whole thing).
   - Your training config YAML for the centered-instance model.
   - The full training/eval log (not just the tail) specifically the precision/recall lines per epoch, like in #2789.
   - If possible, try uploading your `.slp` project to https://slp.sh so we can try to reproduce directly with your actual data (can also upload video here).

Thank you,
Amick
